[wip] Install ec-typescript as a dependecy for addons

ro0gr opened the pull request:
Seems like ember-cli-typescript isn't supposed to work when installed as a dev dep to the host addon. I've just got a blank screen due to missing assets in my case.

In the build output I've seen a warning:

> `ember-cli-typescript` should be included in your `dependencies`, not `devDependencies`

which was great cause it finally led me to a fix. But the question is whether we should convert it to error? Seems like nothing works anyways with `ec-ts` in the addon dev deps..

Also I've found I just wasn't careful enough, cause there are special installation instructions for addon case:

> ember install ember-cli-typescript --save

However, I feel it still can be better if we take care of this installation difference in the blueprint. Which would to allow install this addon by simply running `ember install ember-cli-typescript`.

At the moment I'd like to know if the feature makes sense here?

chriskrycho commented:
We've been EmberConf-ing and traveling back from that; one of us will get back to you in the next few (work) days!

chriskrycho commented:
@ro0gr this seems like a great idea, do you mind rebasing this and we'll iterate forward with you on it?

ro0gr commented:
@chriskrycho sure, will do.

mike-north commented:
Got the warning message today, so now I have a better idea of what we're dealing with here

WARNING: @snap-doc/app has added the decorators plugin to its build, but ember-cli-babel provides these by default now! You can remove the transforms, or the addon that provided them, such as @ember-decorators/babel-transforms. Ember supports the stage 1 decorator spec and transforms, so if you were using stage 2, you'll need to ensure that your decorators are compatible, or convert them to stage 1.

WARNING: @snap-doc/app has added the class-properties plugin to its build, but ember-cli-babel provides these by default now! You can remove the transforms, or the addon that provided them, such as @ember-decorators/babel-transforms.

ro0gr commented:
@mike-north sorry, not sure I follow. Do you suggest to auto-remove "@ember-decorators/babel-transforms" when install "ember-cli-typescript@2"?

chriskrycho commented:
I think he just commented on the wrong issue; that was meant for #640.

ro0gr commented:
I've just updated docs and tests, rebased and squashed. 

According to Azure there is a test failure:

> $ mocha --recursive js/tests -R xunit -O output=node-tests.xml
indexDeclarations is not defined (Error in blueprint template: /home/vsts/work/1/s/node_modules/ember-cli-typescript/blueprint-files/ember-cli-typescript/types/__app_name__/index.d.ts)
ReferenceError: indexDeclarations is not defined (Error in blueprint template: /home/vsts/work/1/s/node_modules/ember-cli-typescript/blueprint-files/ember-cli-typescript/types/__app_name__/index.d.ts)

But that doesn't seem like related one. At least I can see the same failure on a sibling PRs

ro0gr commented on README.md:
````diff
@@ -110,39 +97,13 @@ Follow the same process of deduplication, reinstallation, and re-deduplication a
 ember install ember-cli-typescript@latest
 ```
 
-##### In addons
````
open to suggestions to articulate it more accurate

ro0gr commented on README.md:
````diff
@@ -110,39 +97,13 @@ Follow the same process of deduplication, reinstallation, and re-deduplication a
-    ```sh
-    ember install ember-cli-typescript@latest --save
-    ```
+***Note:* To work properly, starting from v2, ember-cli-typescript must be declared as a `dependency`, not a `devDependency` for addons. With `ember install` this migration will be automatically handled for you.**
````
open to suggestions to articulate it more accurate

chriskrycho replied:
I like what you have here. I would love it if we added back in the earlier instructions, perhaps like so:

```markdown
If you choose to make the upgrade manually with Yarn or npm, here are the steps you need to follow:
```

ro0gr replied:
I've bring non-`ember install` instructions back, but it still requires `ember generate` to work properly. Please check, and let me know if it makes sense to you.

chriskrycho commented:
Thanks again for doing this! On Wednesday (after we've finally fully cleared our backlog of built-up PRs), would you mind rebasing this? We'll coordinate with you to get it polished and integrated! Sorry about the delay. 😅

chriskrycho commented on tests/dummy/app/templates/docs/upgrade-notes.md:
```diff
@@ -44,39 +44,13 @@ Follow the same process of deduplication, reinstallation, and re-deduplication a
 
-### In addons
-
-To work properly, Ember addons must declare this library as a `dependency`, not a `devDependency`. **This is a *change* from ember-cli-typescript v1.**
```
Same comment as in the README.
